pyaccel/optics/miscellaneous.py

```diff
@@ -145,3 +145,10 @@ def get_curlyh(beta, alpha, x, xl):
     """."""
     gamma = (1 + alpha*alpha) / beta
     return beta*xl*xl + 2*alpha*x*xl + gamma*x*x
+
+@_interactive
+def get_emittances_from_coupling(emit0, coupling):
```

Comment:
I don't like this equation. It is not correct if we are dealing with equilibrium parameters.
I think the propagation of such misleading concepts must not be encouraged...
Besides, we are tryng to abolishing the concept of horizontal and vertical emittances, since they don't make much sense.
Why do you need this method, @xresende ?
What do you think @murilobalves ?

Reply:
We often have to use this algebra for splitting an 'emittance' in the two planes, given a (emittance ratio) coupling approximation/estimate. It then made sense for me to define a function for it in our libraries. 

But I agree that this algebra only strictly applies to motion invariants, and within the difference resonance approximation, when the sum of x and y invariants is also an invariant, and 'coupling' is nothing but the ratio of the invariants and 'emit0' is a initial invariant sum value . Nonetheless we often loosely use this algebra to approximate the horizontal and vertical emittance splitting in small-coupling situations, right? For example, when we have a trackcpp model without skew linear fields and want to calculate lifetimes with an ah-hoc coupling parameter value...

are you then suggesting we avoid using this approximation and, for lifetime purpose for example, always work with either a) models with skew quads or b) define ah-hoc emit1 and emit2 parameters (and not, a emit0 and coupling) ?

Reply:
Yes, those options are ok. But we also have a third option which is using the correct approximation formula for emittances under influence of the difference resonance, which takes into account the damping times (in accordance to Chao's paper and Boaz' thesis). 

And, for some rare situation where we, for some reason, want to use this wrong version, we can implement it outside pyaccel, in the specific test or draft script of the occasion.

There was a large effort from our part in the last few months to study coupling and implement methods to estimate equilibrium parameters correctly, with the implementation of ohmi envelope, Edwards and Teng decomposition; and it was king of painful to let some misconceptions go, such as the inexistence of horizontal and vertical emittances, horizontal and vertical tunes, the different concepts of coupling strength, emittances ratio and invariants ratios.

For this reason I think this PR is a step back.

But what was the application you needed to use this formula? Could you implement the correct version? Maybe using some optional parameters for the ratio of the damping times? And considering not an emit0, but initial values for emittances one and two?